enricher/dependency/src/main/java/io/fabric8/maven/enricher/dependency/DependencyEnricher.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2016 Red Hat, Inc.
+ *
+ * Red Hat licenses this file to you under the Apache License, version
+ * 2.0 (the "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package io.fabric8.maven.enricher.dependency;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.KubernetesListBuilder;
+import io.fabric8.maven.core.util.Configs;
+import io.fabric8.maven.enricher.api.BaseEnricher;
+import io.fabric8.maven.enricher.api.EnricherContext;
+import org.apache.maven.artifact.Artifact;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Enricher for adding icons to descriptors
+ *
+ * @author roland
+ * @since 01/05/16
+ */
+public class DependencyEnricher extends BaseEnricher {
+
+  private static String DEPENDENCY_KUBERNETES_YAML = "/META-INF/fabric8/kubernetes.yml";
+  private Set<File> dependencyArtifacts = new HashSet<>();
+
+  public DependencyEnricher(EnricherContext buildContext) {
+    super(buildContext, "dependency");
+
+    Set<Artifact> artifacts = isIncludeTransitive() ?
+        buildContext.getProject().getArtifacts() : buildContext.getProject().getDependencyArtifacts();
+
+    for (Artifact artifact : artifacts) {
+      if (Artifact.SCOPE_COMPILE.equals(artifact.getScope()) && "jar".equals(artifact.getType())) {
+        dependencyArtifacts.add(artifact.getFile());
+      }
+    }
+  }
+
+  @Override
+  public void adapt(KubernetesListBuilder builder) {
+    for (File artifact : dependencyArtifacts) {
+      try {
+        URL url = new URL("jar:file:" + artifact.getAbsolutePath() + "!" + DEPENDENCY_KUBERNETES_YAML);
```

Comment:
Ok, lets leave it like this. Just a bit unsure when `file.getAbsolutePath()` returns a Windows path with `\` whether they work as intended in an URL or whether they should be converted to `/`.  I dont know how hard it would be to write a unit test, so we could verify this when building on Windows.

We might consider to take this over later for the enricher and generator lookup, too.